src/test/java/school/hei/haapi/service/RetakeExamServiceTest.java

```diff
@@ -71,180 +68,87 @@ void crupdateRetakeExams_should_save_all_retake_exams() {
-    expiredSession.setId(sessionId);
-    expiredSession.setDateTo(Instant.now().minus(1, ChronoUnit.DAYS));
-
-    RetakeExam existingRetake = new RetakeExam();
-    existingRetake.setId("existing_retake");
-
-    when(retakeExamSessionService.getById(sessionId)).thenReturn(expiredSession);
-    when(retakeExamRepository.findRetakeExamsBySession_IdAndStudent_Id(sessionId, studentId))
-        .thenReturn(List.of(existingRetake));
-
-    List<RetakeExam> result = retakeExamService.getStudentRetakeExams(sessionId, studentId);
-
-    assertEquals(1, result.size());
-    assertEquals("existing_retake", result.get(0).getId());
-    verify(retakeExamRepository, never())
-        .findRetakeExamsByStudent_IdAndSession_DateFromAfter(anyString(), any());
-  }
-
-  @Test
-  void getStudentRetakeExams_with_active_session_should_generate_new_retake_exams() {
-    String studentId = "student1";
-    String sessionId = "active_session";
-
-    RetakeExamSession activeSession = new RetakeExamSession();
-    activeSession.setId(sessionId);
-    activeSession.setDateTo(Instant.now().plus(1, ChronoUnit.DAYS));
-
-    Course course1 = new Course();
-    course1.setId("course1");
-    Course course2 = new Course();
-    course2.setId("course2");
-
-    CourseResult incompleteCourse1 = new CourseResult();
-    incompleteCourse1.setCourse(course1);
-    incompleteCourse1.setStatus(INCOMPLETE);
-    CourseResult incompleteCourse2 = new CourseResult();
-    incompleteCourse2.setCourse(course2);
-    incompleteCourse2.setStatus(INCOMPLETE);
-
-    YearlyResult yearlyResult = new YearlyResult();
-    yearlyResult.setCourseResults(List.of(incompleteCourse1, incompleteCourse2));
-
-    ResultSummary resultSummary = new ResultSummary();
-    resultSummary.setYearlyResults(List.of(yearlyResult));
-
-    when(retakeExamSessionService.getById(sessionId)).thenReturn(activeSession);
-    when(retakeExamRepository.findRetakeExamsBySession_IdAndStudent_Id(sessionId, studentId))
-        .thenReturn(List.of());
-    when(retakeExamRepository.findRetakeExamsByStudent_IdAndSession_DateFromAfter(
-            anyString(), any()))
-        .thenReturn(List.of());
-    when(gradeResultService.getStudentResultSummary(studentId)).thenReturn(resultSummary);
-
-    school.hei.haapi.model.Course domainCourse1 = new school.hei.haapi.model.Course();
-    domainCourse1.setId("course1");
-    school.hei.haapi.model.Course domainCourse2 = new school.hei.haapi.model.Course();
-    domainCourse2.setId("course2");
-    when(courseMapper.toDomain(course1)).thenReturn(domainCourse1);
-    when(courseMapper.toDomain(course2)).thenReturn(domainCourse2);
-
-    List<RetakeExam> result = retakeExamService.getStudentRetakeExams(sessionId, studentId);
-
-    assertNotNull(result);
-    assertEquals(2, result.size());
-  }
+  void getAllStudentRetakeExams_with_two_future_sessions() {
+    school.hei.haapi.model.User student1 = new school.hei.haapi.model.User();
```

Comment:
```suggestion
    var student1 = new school.hei.haapi.model.User();
```

use `var` when right hand side type is _OBVIOUS_

Reply:
yeah